pkg/controller/pravegacluster/pravegacluster_controller.go

```diff
@@ -239,12 +239,20 @@ func (r *ReconcilePravegaCluster) deploySegmentStore(p *pravegav1alpha1.PravegaC
 		return err
 	}
 
+	cm := &corev1.ConfigMap{}
+	name := p.Spec.Pravega.SegmentStoreEnvVars
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: p.Namespace}, cm)
 	configMap := pravega.MakeSegmentstoreConfigMap(p)
 	controllerutil.SetControllerReference(p, configMap, r.scheme)
 	err = r.client.Create(context.TODO(), configMap)
 	if err != nil && !errors.IsAlreadyExists(err) {
 		return err
 	}
+	flag := r.compareKeys(cm, configMap)
```

Comment:
Could we create a new method `validateUserConfigMap()` and do the get on user supplied config map and compareKeys inside the same?